google/cloud/bigtable/internal/table.h

```diff
@@ -149,12 +150,9 @@ class Table {
         "The arguments passed to ReadModifyWriteRow(row_key,...) must be "
         "convertible to bigtable::ReadModifyWriteRule");
 
-    // TODO(#336) - optimize this code by not copying the parameter pack.
-    // Add first default rule
-    *request.add_rules() = rule.as_proto_move();
     // Add if any additional rule is present
     std::initializer_list<bigtable::ReadModifyWriteRule> rule_list{
-        std::forward<Args>(rules)...};
+        rule, std::forward<Args>(rules)...};
```

Comment:
This actually introduces an extra copy of the `rule` parameter.  `rule_list` copies all its arguments.